src/app/modules/core/core.errors.ts

```diff
@@ -32,3 +32,9 @@ export class DatabaseError extends ApplicationError {
     super(message)
   }
 }
+
+export class ExternalError extends ApplicationError {
```

Comment:
is it a good idea to have a generic `ExternalError` though? we have so many different kinds of interactions with different services, I'm not sure this abstraction is useful. eg what if a service interacts with two different external services, but the controller above it needs to be able to differentiate between them?

Reply:
I'm not sure about the alternative, since we need our own casted error to let the controllers know what's up

Reply:
exactly, so the alternative would be to have a custom `CreatePresignedUrlError` instead of a generic `ExternalError`

Reply:
Noted. Will refactor that in #607